app/src/pages/Play/YourTurnContent.tsx

```diff
@@ -358,29 +358,34 @@ function YourTurnContent(props: {
                     props.secondsLeft !== 0
 
                   const scorings = compact(
-                    shownCardIds.map((cardId) => {
-                      const card = shownCardsInActiveTurn.get(cardId)
-                      if (card) {
-                        return {
-                          turn_id: props.activeTurn.id,
-                          card_id: cardId,
-                          score:
-                            card.status === ShownCardStatus.Complete ? 1 : 0,
-                          status: card.status,
-                          started_at: timestamptzNowFromDate(card.startedAt),
-                          ended_at: timestamptzNowFromDate(card.endedAt),
+                    shownCardIds.map<TurnScoringsInsertInput | null>(
+                      (cardId) => {
+                        const card = shownCardsInActiveTurn.get(cardId)
+                        if (card) {
+                          return {
+                            turn_id: props.activeTurn.id,
+                            card_id: cardId,
+                            score:
+                              card.status === ShownCardStatus.Complete ? 1 : 0,
+                            status: card.status,
+                            started_at: new Date(
+                              card.startedAt.getTime() + props.serverTimeOffset
+                            ),
+                            ended_at: new Date(
+                              card.endedAt.getTime() + props.serverTimeOffset
+                            ),
```

Comment:
This is the only remaining context in which we're setting timestamps via the client's clock – and it's now adjusted for clock drift. It also probably matters less given this is scoring stuff in which the _difference_ between these timestamps is really what matters.

Reply:
I suppose the better way to do this, is to not batch these insertion requests, and instead asynchronously insert the turn scoring at the time "next" or "skip" is clicked. And at the review stage, the turn scorings can be adjusted (tho not the start and end time).

Reply:
Yeah, good point – although probably not the end of the world in its current state / something we can followup with if we find it worthwhile?